multiqc/modules/base_module.py

```diff
@@ -482,6 +486,32 @@ def add_data_source(self, f=None, s_name=None, source=None, module=None, section
         except AttributeError:
             logger.warning("Tried to add data source for {}, but was missing fields data".format(self.name))
 
+    def add_software_version(self, version: str, sample: str = None):
+        """Save software versions for module."""
+        # Don't add if sample is ignored
+        if sample is not None and self.is_ignore_sample(sample):
+            return
+
+        # Check if version string is PEP 440 compliant to enable version normalization and proper ordering.
+        # Otherwise use raw string is used for version.
+        # - https://peps.python.org/pep-0440/
+        # Use flag `is_compliant` so we don't log the same warning multiple times.
+        version, is_compliant = software_versions.parse_version(version)
+
+        if version in self.versions:
+            return
+
+        if is_compliant:
+            logger.debug(f"Version '{version}' in module {self.name} does not conform to PEP 440 format")
```

Comment:
Honestly I'm not sure that we really need a debug log here at all. I think this will happen a lot and I wonder if it'll make people worry if they see it in the log.

```suggestion
            logger.debug(f"Version '{version}' in module {self.name} does not conform to PEP 440 format - cannot sort list of versions nicely")
```

Reply:
Removed in 29f82cf45f83e1bc5095fc9b6e932a872f81fea6